diff -Naur mpc-0.5/configure mpc-0.5-patched/configure
--- mpc-0.5/configure	2008-09-17 16:53:49.000000000 +0000
+++ mpc-0.5-patched/configure	2008-09-26 08:06:16.000000000 +0000
@@ -18737,13 +18737,13 @@
 fi
 
 # Checks for libraries.
-echo "$as_me:$LINENO: checking for mpfr_clear in -lmpfr" >&5
-echo $ECHO_N "checking for mpfr_clear in -lmpfr... $ECHO_C" >&6
-if test "${ac_cv_lib_mpfr_mpfr_clear+set}" = set; then
+echo "$as_me:$LINENO: checking for __gmpz_init in -lgmp" >&5
+echo $ECHO_N "checking for __gmpz_init in -lgmp... $ECHO_C" >&6
+if test "${ac_cv_lib_gmp___gmpz_init+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lmpfr  $LIBS"
+LIBS="-lgmp  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -18757,11 +18757,11 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char mpfr_clear ();
+char __gmpz_init ();
 int
 main ()
 {
-mpfr_clear ();
+__gmpz_init ();
   ;
   return 0;
 }
@@ -18788,34 +18788,34 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_mpfr_mpfr_clear=yes
+  ac_cv_lib_gmp___gmpz_init=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_mpfr_mpfr_clear=no
+ac_cv_lib_gmp___gmpz_init=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_mpfr_mpfr_clear" >&5
-echo "${ECHO_T}$ac_cv_lib_mpfr_mpfr_clear" >&6
-if test $ac_cv_lib_mpfr_mpfr_clear = yes; then
-  LIBS="-lmpfr $LIBS"
+echo "$as_me:$LINENO: result: $ac_cv_lib_gmp___gmpz_init" >&5
+echo "${ECHO_T}$ac_cv_lib_gmp___gmpz_init" >&6
+if test $ac_cv_lib_gmp___gmpz_init = yes; then
+  LIBS="-lgmp $LIBS"
 else
-  { { echo "$as_me:$LINENO: error: libmpfr not found or uses a different ABI." >&5
-echo "$as_me: error: libmpfr not found or uses a different ABI." >&2;}
+  { { echo "$as_me:$LINENO: error: libgmp not found or uses a different ABI." >&5
+echo "$as_me: error: libgmp not found or uses a different ABI." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
-echo "$as_me:$LINENO: checking for __gmpz_init in -lgmp" >&5
-echo $ECHO_N "checking for __gmpz_init in -lgmp... $ECHO_C" >&6
-if test "${ac_cv_lib_gmp___gmpz_init+set}" = set; then
+echo "$as_me:$LINENO: checking for mpfr_clear in -lmpfr" >&5
+echo $ECHO_N "checking for mpfr_clear in -lmpfr... $ECHO_C" >&6
+if test "${ac_cv_lib_mpfr_mpfr_clear+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lgmp  $LIBS"
+LIBS="-lmpfr  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -18829,11 +18829,11 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char __gmpz_init ();
+char mpfr_clear ();
 int
 main ()
 {
-__gmpz_init ();
+mpfr_clear ();
   ;
   return 0;
 }
@@ -18860,24 +18860,24 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_gmp___gmpz_init=yes
+  ac_cv_lib_mpfr_mpfr_clear=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_gmp___gmpz_init=no
+ac_cv_lib_mpfr_mpfr_clear=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_gmp___gmpz_init" >&5
-echo "${ECHO_T}$ac_cv_lib_gmp___gmpz_init" >&6
-if test $ac_cv_lib_gmp___gmpz_init = yes; then
-  LIBS="-lgmp $LIBS"
+echo "$as_me:$LINENO: result: $ac_cv_lib_mpfr_mpfr_clear" >&5
+echo "${ECHO_T}$ac_cv_lib_mpfr_mpfr_clear" >&6
+if test $ac_cv_lib_mpfr_mpfr_clear = yes; then
+  LIBS="-lmpfr $LIBS"
 else
-  { { echo "$as_me:$LINENO: error: libgmp not found or uses a different ABI." >&5
-echo "$as_me: error: libgmp not found or uses a different ABI." >&2;}
+  { { echo "$as_me:$LINENO: error: libmpfr not found or uses a different ABI." >&5
+echo "$as_me: error: libmpfr not found or uses a different ABI." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
diff -Naur mpc-0.5/configure.ac mpc-0.5-patched/configure.ac
--- mpc-0.5/configure.ac	2008-09-16 16:23:57.000000000 +0000
+++ mpc-0.5-patched/configure.ac	2008-09-26 08:06:32.000000000 +0000
@@ -47,12 +47,12 @@
 fi
 
 # Checks for libraries.
-AC_CHECK_LIB([mpfr], [mpfr_clear],
-             [LIBS="-lmpfr $LIBS"],
-             [AC_MSG_ERROR([libmpfr not found or uses a different ABI.])])
 AC_CHECK_LIB([gmp], [__gmpz_init],
              [LIBS="-lgmp $LIBS"],
              [AC_MSG_ERROR([libgmp not found or uses a different ABI.])])
+AC_CHECK_LIB([mpfr], [mpfr_clear],
+             [LIBS="-lmpfr $LIBS"],
+             [AC_MSG_ERROR([libmpfr not found or uses a different ABI.])])
 
 # Checks for header files.
 AC_HEADER_STDC
